src/crypto.ts

```diff
@@ -195,4 +200,65 @@ export default class Crypto {
       decodeBase64(formSecretKey)
     )
   }
+
+  /**
+   * Decrypts an encrypted submission, and also download and decrypt any attachments alongside it.
+   * @param formSecretKey Secret key as a base-64 string
+   * @param decryptParams The params containing encrypted content and information.
+   * @returns A promise of the decrypted submission, including attachments (if any). Or else returns null if a decryption error decrypting any part of the submission.
+   * @throws {MissingPublicKeyError} if a public key is not provided when instantiating this class and is needed for verifying signed content.
+   */
+  decryptWithAttachments = async (
+    formSecretKey: string,
+    decryptParams: DecryptParams
+  ): Promise<DecryptedContentAndAttachments | null> => {
+    const decryptedRecords: DecryptedAttachments = {}
+    const filenames: Record<string, string> = {}
+
+    const attachmentRecords: EncryptedAttachmentRecords = decryptParams.attachmentDownloadUrls ?? {}
+    const decryptedContent = this.decrypt(formSecretKey, decryptParams)
+    if (decryptedContent === null) return null
+
+    // Retrieve all original filenames for attachments for easy lookup
+    decryptedContent.responses.forEach((response) => {
+      if (response.fieldType === 'attachment' && response.answer) {
+        filenames[response._id] = response.answer
+      }
+    })
+
+    const downloadPromises : Array<Promise<void>> = []
+    for (let fieldId in attachmentRecords) {
+      // Original name for the file is not found
+      if (filenames[fieldId] === undefined) return null
+
+      downloadPromises.push(
+        axios.get(attachmentRecords[fieldId], { responseType: 'json' })
+      .then((downloadResponse) => {
+        if (downloadResponse.status !== 200) throw new Error("Download failed")
```

Comment:
I don't think this line is necessary. if the response status code is outside [200, 300), Axios will reject the promise by default